src/common/state/redis.c

```diff
@@ -886,11 +909,22 @@ void redis_db_client_table_subscribe_callback(redisAsyncContext *c,
    * client_type string, and we add 1 to null-terminate the string. */
   int client_type_length = payload->len - 1 - sizeof(client.id) + 1;
   char *client_type = malloc(client_type_length);
-  memcpy(client_type, &payload->str[1 + sizeof(client.id)], client_type_length);
+  char *aux_address = malloc(client_type_length);
```

Comment:
The following code is very confusing, is it possible to use strtok or strsep to split the string and then use strdup to copy the relevant part?